tests/sweep_framework/sweeps/ccl/all_gather_t3k_non_sharded.py

```diff
@@ -0,0 +1,354 @@
+        "input_shape": row_input_shapes,
+        "dim": [0, 1, 2, 3],
+        "tensor_layout": [ttnn.ROW_MAJOR_LAYOUT],
+        "input_dtype": [ttnn.bfloat16],
+        "mem_config": [
+            ttnn.MemoryConfig(buffer_type=ttnn.BufferType.L1),
+        ],
+        "all_gather_operation": ["line_all_gather"],
+        "num_iters": [1],
+    },
+    "all_gather_row_non_sharded_l1_line_bf8": {
+        "num_devices": [4, 8],
+        "num_links": [1, 2],
+        "input_shape": row_input_shapes,
+        "dim": [0, 1, 2, 3],
+        "tensor_layout": [ttnn.ROW_MAJOR_LAYOUT],
+        "input_dtype": [ttnn.bfloat8_b],
+        "mem_config": [
+            ttnn.MemoryConfig(buffer_type=ttnn.BufferType.L1),
+        ],
+        "all_gather_operation": ["line_all_gather"],
+        "num_iters": [1],
+    },
+}
+
+
+def invalidate_vector(test_vector) -> Tuple[bool, Optional[str]]:
+    unchunked_input_shape = list(test_vector["input_shape"])
+    unchunked_input_shape[test_vector["dim"]] *= test_vector["num_devices"]
+    if test_vector["num_devices"] < 2:
+        return True, f"Requires multiple devices to run"
+
+    if unchunked_input_shape[test_vector["dim"]] % test_vector["num_devices"] != 0 or (
+        test_vector["dim"] == 3 and unchunked_input_shape[test_vector["dim"]] // test_vector["num_devices"] % 32 != 0
+    ):
+        return True, f"Unsupported test case"
+    if test_vector["tensor_layout"] == ttnn.ROW_MAJOR_LAYOUT and test_vector["input_dtype"] == ttnn.bfloat8_b:
+        return True, f"bfloat8_b/4_b only supports TILE layout"
+    if test_vector["num_links"] == 2 and test_vector["num_devices"] == 8:
+        return True, f"8 devices and 2 links are not supported on t3000 devices"
+    return False, None
+
+
+def mesh_device_fixture():
+    assert ttnn.get_num_devices() >= 8, "Not T3000!"
+    device_ids = [0, 4, 5, 1, 2, 6, 7, 3]
+    num_devices_requested = len(device_ids)
+    mesh_device = ttnn.open_mesh_device(ttnn.MeshShape(1, num_devices_requested), device_ids[:num_devices_requested])
+    print("ALL GATHER: Opened device mesh")
+
+    yield (mesh_device, "T3000 Mesh")
+
+    print("ALL GATHER: Closing device mesh")
+    for device in mesh_device.get_devices():
+        ttnn.DumpDeviceProfiler(device)
+    ttnn.close_mesh_device(mesh_device)
+    del mesh_device
+
+
+# This is the run instructions for the test, defined by the developer.
+# The run function must take the above-defined parameters as inputs.
+# The runner will call this run function with each test vector, and the returned results from this function will be stored.
+
+
+def run(
+    num_devices,
+    input_shape,
+    dim,
+    num_links,
+    input_dtype,
+    tensor_layout,
+    mem_config,
+    # num_cores,
+    num_iters,
+    all_gather_operation,
+    *,
+    device,
+):
+    t3k_mesh_device = device
+
+    logger.info(f"Input shape: {input_shape}")
+    logger.info(f"dim: {dim}")
+
+    input_tensor = torch.rand(input_shape).bfloat16()
+
```

Comment:
It looks like the golden function should be the identity function here.  In other words, we pass this input_tensor to the golden function (just an identity) and then return it back.  I believe we want to continue to push the use of the golden function even in this case.

Reply:
@eyonland golden function are planned to track as a separate issue and not to mingle with this. Tracked from #12747